packages/mitex/specs/README.md

```diff
@@ -0,0 +1,314 @@
+
+Even if you don't know Rust at all, you can still add missing TeX commands to MiTeX by modifing [latex-std](./latex/standard.typ), since they are written in typst! You can open an issue to acquire the commands you want to add, or you can edit the files and submit a pull request.
+
+
+## Concepts
+
+Currently, MiTeX maintains following three parts of code:
+
+- A TeX parser library written in **Rust**, see [mitex-lexer](./crates/mitex-lexer) and [mitex-parser](./crates/mitex-parser).
+- A TeX to Typst converter library written in **Rust**, see [mitex](./crates/mitex).
+- A list of TeX packages and comamnds written in **Typst**, which then used by the typst package, see [MiTeX Command Specification](./packages/mitex/specs).
+
+For a translation process, for example, we have:
+
+```
+\frac{1}{2}
+
+===[parser]===> AST ===[converter]===>
+
+#eval("$frac(1, 2)$", scope: (frac: (num, den) => $(num)/(den)$))
+```
+
+You can use the `\mitex-convert()` function to get the Typst Code generated from LaTeX Code.
+
+To achieve this, we need to define four components for LaTeX commands:
+
+- `cmd`: The name of the LaTeX command. Since LaTeX commands all start with `\`, we remove the leading `\` and use the command name as the key in the dictionary.
+  - There is also the concept of environments, for example, `\begin{pmatrix} 1 & 2 \\ 3 & 4 \end{pmatrix}` is a matrix environment.
+- `args`: The argument pattern of the LaTeX command. For example:
+  - `\alpha` has no arguments;
+  - `\hat{x}` matches one argument on the right;
+  - `\frac{1}{2}` matches two arguments on the right;
+  - `\sqrt[3]{2}` includes optional arguments;
+  - `\sum\limits` has `limits` matching one argument on the left;
+  - `\displaystyle` greedily matches all arguments on the right;
+  - `x \over y` as an infix operator greedily matches all arguments on both sides.
+- `alias`: The alias of the LaTeX command in Typst.
+  - The alias can be an existing symbol or function, for example, `alpha` and `binom(n, k)`;
+  - It can also be a key in the `mitex-scope` for `eval`, for example, our self-defined `frac`.
+- `handle`: The value in the `mitex-scope`, which is our self-defined symbol or processing function.
+  - For example, the `frac` key corresponds to the value `(num, den) => $(num)/(den)$`.
+
+The parser only needs `cmd` and `args` to generate an AST, while the converter requires `cmd`, `args`, and `alias` to generate the corresponding Typst code. Since both the parser and converter are part of the WASM plugin written in Rust, we refer to the dictionary composed of `cmd`, `args`, and `alias` as spec and pre-compile it into the wasm.
+
+At the Typst level, we need `alias` and `handle`, which are then combined into the `mitex-scope` passed to the `eval` function as the `scope` parameter.
+
+Therefore, we need to define the spec here, and later we will also support custom spec.
+
+## Reference
```

Comment:
In short, currently a _command spec_ is a typst dict, with its keys are name of the tex command, and its values are _opaque spec item_, which contains necessary information for MiTeX.

In [specification file for standard latex](./latex/standard.typ), It constructs and exports a such _command spec_.

Next, we provides a bunch of convenient functions for constructing _opaque spec items_.